core/src/main/java/org/apache/iceberg/PositionDeletesTable.java

```diff
@@ -0,0 +1,366 @@
+    @Override
+    public long sizeBytes() {
+      return length();
+    }
+
+    String schemaString() {
+      return schemaString;
+    }
+
+    String specString() {
+      return specString;
+    }
+
+    ResidualEvaluator evaluator() {
+      return evaluator;
+    }
+
+    Types.StructType partitionType() {
+      return partitionType;
+    }
+
+    @Override
+    protected ScanTask self() {
+      return this;
+    }
+
+    @Override
+    protected ScanTask newSplitTask(ScanTask parentTask, long offset, long length) {
+      return new SplitScanTask(parentTask, offset, length);
+    }
+
+    /**
+     * Utility method to get constant values of rows to be scanned by this task. The following
+     * columns are constants of each PositionDeletesFileScanTask: * spec_id * partition *
+     * delete_file_path
+     *
+     * @return a map of column id to constant values returned by this task type
+     */
+    public Map<Integer, ?> constantsMap() {
+      return constantsMap((type, constant) -> constant);
+    }
+
+    /**
+     * Utility method to get constant values of rows to be scanned by this task. The following
+     * columns are constants of each PositionDeletesFileScanTask: * spec_id * partition *
+     * delete_file_path
+     *
+     * @param convertConstant callback to convert from an Iceberg typed object to an engine specific
+     *     object
+     * @return a map of column id to constant values returned by this task type
+     */
+    public Map<Integer, ?> constantsMap(BiFunction<Type, Object, Object> convertConstant) {
+      Map<Integer, Object> idToConstant = Maps.newHashMap();
+      StructLike partitionData = file().partition();
+
+      idToConstant.put(
+          POSITION_DELETE_TABLE_FILE_PATH,
+          convertConstant.apply(Types.StringType.get(), file().path()));
+
+      // add spec_id constant column
+      idToConstant.put(
+          POSITION_DELETE_TABLE_SPEC_ID,
+          convertConstant.apply(Types.IntegerType.get(), file().specId()));
+
+      // add partition constant column
+      if (partitionType != null) {
+        if (partitionType.fields().size() > 0) {
+          StructLike coercedPartition =
+              PartitionUtil.coercePartition(partitionType, spec(), partitionData);
+          idToConstant.put(
+              POSITION_DELETE_TABLE_PARTITION_FIELD_ID,
+              convertConstant.apply(partitionType, coercedPartition));
+        } else {
+          // use null as some query engines may not be able to handle empty structs
+          idToConstant.put(POSITION_DELETE_TABLE_PARTITION_FIELD_ID, null);
+        }
+      }
+
+      return idToConstant;
+    }
+  }
+
+  public static class SplitScanTask extends ScanTask
+      implements ContentScanTask<DeleteFile>, MergeableScanTask<SplitScanTask> {
+
+    private final ScanTask parentTask;
+    private final long offset;
+    private final long len;
+
+    protected SplitScanTask(ScanTask parentTask, long offset, long length) {
+      super(
```

Comment:
Note: SplitScanTask needs to be an instance of ScanTask because ScanTask.newSplitTask needs to return ScanTask to satisfy the type.